src/utils/time.ts

```diff
@@ -90,45 +90,65 @@ export function getDateTimeString(timestamp: number) {
  * @param stage - Current proposal stage
  * @param proposalTimestamp - Proposal's base timestamp (queue time for Queued, dequeue time for others)
  */
-export function getEndHumanEndTime({
-  proposalTimestamp,
+export function getHumanEndTime({
+  queuedAt,
+  dequeuedAt,
+  executedAt,
   stage,
+  quorumMet,
 }: {
-  proposalTimestamp: number | undefined;
+  queuedAt: string | null;
+  dequeuedAt: string | null;
+  executedAt: string | null;
   stage: ProposalStage | undefined;
+  quorumMet: boolean | null;
 }): string | undefined {
   const now = Date.now();
 
-  if (!proposalTimestamp || !stage) {
+  const hasAnyTS = queuedAt || dequeuedAt || executedAt;
+  if (!stage || !hasAnyTS) {
     return undefined;
   }
 
   switch (stage) {
     case ProposalStage.Queued: {
-      const endDate = getStageEndTimestamp(stage, proposalTimestamp)!;
+      const endDate = getStageEndTimestamp(stage, new Date(queuedAt!).getTime())!;
       return `Expires in ${getHumanReadableDuration(endDate - now)} on ${getFullDateHumanDateString(endDate)}`;
     }
     case ProposalStage.Referendum: {
-      const endDate = getStageEndTimestamp(stage, proposalTimestamp)!;
+      const endDate = getStageEndTimestamp(stage, new Date(dequeuedAt!).getTime())!;
       return `Voting ends in ${getHumanReadableDuration(endDate - now)} on ${getFullDateHumanDateString(endDate)}`;
     }
     case ProposalStage.Approval:
     // DEPRECATED: Treat like Execution (awaiting execution after approval)
     // Fall through to Execution case
     case ProposalStage.Execution: {
-      const endDate = getStageEndTimestamp(stage, proposalTimestamp);
+      const endDate = getStageEndTimestamp(stage, new Date(dequeuedAt!).getTime());
       if (endDate) {
         return `Execution window ends in ${getHumanReadableDuration(endDate - now)} on ${getFullDateHumanDateString(endDate)}`;
       }
       return 'Awaiting execution';
     }
+    case ProposalStage.Rejected: {
+      const endDate = getStageEndTimestamp(
+        ProposalStage.Referendum,
+        new Date(dequeuedAt!).getTime(),
+      );
+      return `Rejected ${getHumanReadableTimeString(endDate!)}`;
+    }
     case ProposalStage.Withdrawn:
```

Comment:
oh withdrawn will actually have an unknown end time. we should not treat it the same as expired. 

almost certain that withdrawn proposals never made it onchain. (according the governance repo readme at least)